http_server/main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+)
+
+type userData struct {
+	username string
+	password string
+	email    string
+}
+
+var (
+
+	// Maps usernames to user's passwords
+	usersMap = make(map[string]userData)
+	// Maps sessions to usernames
+	sessionMap = make(map[string]string)
+)
+
+func welcomeHandler(w http.ResponseWriter, r *http.Request) {
+	names, ok := r.URL.Query()["name"]
+
+	if !ok || len(names[0]) < 1 {
+		http.Error(w, "400 Bad Request", 400)
+		return
+	}
+
+	name := names[0]
+	fmt.Fprintf(w, "<h1>Hello, %s!</h1>", name)
```

Comment:
Going to leave this comment just once, as we can focus on this in subsequent commits: use `template/html` instead.

Here and everywhere else where responses are generated.